openedx_tagging/core/tagging/import_export/api.py

```diff
@@ -0,0 +1,184 @@
+            )
+        )
+
+    # Creating import task
+    task = TagImportTask.create(taxonomy)
+
+    try:
+        # Get the parser and parse the file
+        task.log_parser_start()
+        parser = get_parser(parser_format)
+        tags, errors = parser.parse_import(file)
+
+        # Check if there are errors in the parse
+        if errors:
+            task.handle_parser_errors(errors)
+            return False
+
+        task.log_parser_end()
+
+        # Generate actions
+        task.log_start_planning()
+        tag_import_plan = TagImportPlan(taxonomy)
+        tag_import_plan.generate_actions(tags, replace)
+        task.log_plan(tag_import_plan)
+
+        if tag_import_plan.errors:
+            task.handle_plan_errors()
+            return False
+
+        task.log_start_execute()
+        tag_import_plan.execute(task)
+        task.end_success()
+        return True
+    except Exception as exception:
+        # Log any exception
+        task.log_exception(exception)
+        return False
+
+
+def get_last_import_status(taxonomy: Taxonomy) -> TagImportTaskState:
+    """
+    Get status of the last import task of the given taxonomy
+    """
+    task = _get_last_import_task(taxonomy)
+    return task.status
+
+
+def get_last_import_log(taxonomy: Taxonomy) -> str:
+    """
+    Get logs of the last import task of the given taxonomy
+    """
+    task = _get_last_import_task(taxonomy)
+    return task.log
+
+
+def export_tags(taxonomy: Taxonomy, output_format: ParserFormat) -> str:
+    """
+    Returns a string with all tag data of the given taxonomy
+    """
+    _import_export_validations(taxonomy)
+    parser = get_parser(output_format)
+    return parser.export(taxonomy)
+
+
+def _check_unique_import_task(taxonomy: Taxonomy) -> bool:
+    """
+    Verifies if there is another in progress import task for the
+    given taxonomy
+    """
+    last_task = _get_last_import_task(taxonomy)
+    if not last_task:
+        return True
+    return (
+        last_task.status == TagImportTaskState.SUCCESS.value
+        or last_task.status == TagImportTaskState.ERROR.value
+    )
+
+
+def _get_last_import_task(taxonomy: Taxonomy) -> TagImportTask:
+    """
+    Get the last import task for the given taxonomy
+    """
+    return (
+        TagImportTask.objects.filter(taxonomy=taxonomy)
+        .order_by("-creation_date")
+        .first()
+    )
+
+
+def _import_export_validations(taxonomy: Taxonomy):
+    """
+    Validates if the taxonomy is allowed to import or export tags
+    """
+    taxonomy = taxonomy.cast()
+    if taxonomy.allow_free_text:
+        raise ValueError(
+            _(
+                f"Invalid taxonomy ({taxonomy.id}): You cannot import/export a free-form taxonomy."
+            )
+        )
+    if taxonomy.system_defined:
+        raise ValueError(
+            _(
+                f"Invalid taxonomy ({taxonomy.id}): You cannot import/export a system-defined taxonomy."
```

Comment:
@bradenmacdonald I thinks is a combination of A and C. If we call `export` in `AuthorTaxonomy`, it will return a list names and usernames, in other words it would be an API to read users in the system. About `import`, this functionality does not create new taxonomies, it uses already created taxonomies. System-defined taxonomies already created should not be editable via import

Reply:
OK, that makes sense. Thanks.